generators/server/files.js

```diff
@@ -99,14 +107,24 @@ const serverFiles = {
             ]
         },
         {
-            condition: generator => generator.databaseType === 'sql',
+            condition: generator => generator.databaseType === 'sql' || generator.databaseType === 'oauth2',
             path: SERVER_TEST_SRC_DIR,
             templates: [
                 {
                     file: 'package/TestResources.java',
                     renameTo: generator => `${generator.javaDir}/TestResources.java`
                 }
             ]
+        },
+        {
+            condition: generator => generator.authenticationType === 'oauth2',
+            path: SERVER_TEST_SRC_DIR,
+            templates: [
+                {
+                    file: 'package/KeycloakServerTestResource.java',
```

Comment:
We shouldn't have any classes with an identity provider in its name because it should be easy to switch from one to another without changing any code. For Spring Security, we mock the provider in a [`TestSecurityConfiguration`](https://github.com/jhipster/generator-jhipster/blob/main/generators/server/templates/src/test/java/package/config/TestSecurityConfiguration.java.ejs) class.